src/java.base/share/classes/jdk/internal/foreign/CABI.java

```diff
@@ -83,20 +82,16 @@ private static CABI computeCurrent() {
                     // The Linux ABI follows the standard AAPCS ABI
                     return LINUX_AARCH_64;
                 }
-            } else if (arch.equals("ppc64le")) {
+            } else if (arch.startsWith("ppc64")) {
                 if (OperatingSystem.isLinux()) {
                     return LINUX_PPC_64_LE;
+                } else {
+                    return AIX_PPC_64;
                 }
             } else if (arch.equals("riscv64")) {
                 if (OperatingSystem.isLinux()) {
                     return LINUX_RISCV_64;
                 }
-            } else if (arch.startsWith("ppc64")) {
-                if (OperatingSystem.isLinux()) {
-                    return SYS_V_PPC_64LE;
-                } else {
-                    return AIX_PPC_64;
-                }
             } else if (arch.equals("s390x") && OperatingSystem.isLinux()) {
                 return SYS_V_S390X;
             }
```

Comment:
I think we should keep the AIX case separate instead of changing line 85 as compared with upstream:
```java
            } else if (arch.equals("ppc64")) {
                if (OperatingSystem.isAix()) {
                    return AIX_PPC_64;
                }
            } else if (arch.equals("ppc64le")) {
                if (OperatingSystem.isLinux()) {
                    return LINUX_PPC_64_LE;
                }
            } else if (arch.equals("riscv64")) {
                if (OperatingSystem.isLinux()) {
                    return LINUX_RISCV_64;
                }
            } else if (arch.equals("s390x") && OperatingSystem.isLinux()) {
                return SYS_V_S390X;
            }
```

Reply:
Agreed (which was my initial thought but I changed it later for simplicity) and updated as suggested above.